pylib/spinnaker/codelab_config.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from spinnaker.yaml_util import YamlBindings
+
+def configure_codelab_igor_jenkins():
+    """Configures Igor to be enabled and to point to the codelab jenkins instance.
+
+    """
+    YamlBindings.update_yml_source(
+      '/opt/spinnaker/config/spinnaker-local.yml',
+      {
+        'jenkins': {
+          'defaultMaster': {
+            'name': 'CodelabJenkins',
+            'baseUrl': 'http://localhost:9090',
+            'name': 'admin',
+            'name': 'admin'
```

Comment:
name is twice. One of these is probably password. I think password should be injected in as a commandline argument so you can generate it at runtime for unique and private access only to the user. You dont need to worry about generation here, but the caller would generate the password, attach it to the instance metadata, then pass it to this script. The user would learn the password looking at the instance metadata. That's how the marketplace instances seem to work.